hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OmTransportFactory.java

```diff
@@ -34,26 +37,37 @@ OmTransport createOmTransport(ConfigurationSource source,
 
   static OmTransport create(ConfigurationSource conf,
       UserGroupInformation ugi, String omServiceId) throws IOException {
-    OmTransportFactory factory = createFactory();
+    OmTransportFactory factory = createFactory(conf);
 
     return factory.createOmTransport(conf, ugi, omServiceId);
   }
 
-  static OmTransportFactory createFactory() throws IOException {
-    ServiceLoader<OmTransportFactory> transportFactoryServiceLoader =
-        ServiceLoader.load(OmTransportFactory.class);
-    Iterator<OmTransportFactory> iterator =
-        transportFactoryServiceLoader.iterator();
-    if (iterator.hasNext()) {
-      return iterator.next();
-    }
+  static OmTransportFactory createFactory(ConfigurationSource conf)
+      throws IOException {
     try {
+      // if configured transport class is different than the default
+      // Hadoop3OmTransportFactory, then check service loader for
+      // transport class and instantiate it
+      if (conf
+          .get(OZONE_OM_TRANSPORT_CLASS,
+              OZONE_OM_TRANSPORT_CLASS_DEFAULT) !=
+          OZONE_OM_TRANSPORT_CLASS_DEFAULT) {
```

Comment:
This check is not very robust. If the default transport factory is changed, then it would still end up loading Hadoop3OmTransportFactory instead of the new default.

Reply:
Right, the OmTransportFactory was hardcoded to instantiate the Hadoop3OmTransportFactory implementation when no other implementation was provided.  I've updated this to using the default, `OZONE_OM_TRANSPORT_CLASS_DEFAULT`, when no other as suggested.  Thanks @hanishakoneru, all changes pushed in latest commit.